src/components/TokenSelect/index.tsx

```diff
@@ -129,7 +131,20 @@ export default memo(function TokenSelect({ value, collapsed, disabled, onSelect
   usePrefetchBalances()
 
   const [open, setOpen] = useState(false)
-  const onOpen = useCallback(() => setOpen(true), [])
+  const onTokenSelectorClick = useAtomValue(onTokenSelectorClickAtom)
+  const onOpen = useCallback(() => {
+    const promise = onTokenSelectorClick?.()
+    if (promise) {
+      return promise
+        .then((open) => {
+          setOpen(open)
+        })
+        .catch(() => {
+          setOpen(false)
+        })
+    }
+    return setOpen(true)
+  }, [])
```

Comment:
call the handler if it exists

if we get a promise from the handler, then use its resolved value to set the token selector open state

if the promise rejects, then don't open the token selector

otherwise set the open state to true